storage/tokudb/ha_tokudb.h

```diff
@@ -1037,15 +1037,28 @@ class ha_tokudb : public handler {
 #if TOKU_INCLUDE_WRITE_FRM_DATA
     int write_frm_data(const uchar *frm_data, size_t frm_len);
 #endif
-#if TOKU_INCLUDE_UPSERT
 private:
-    int fast_update(THD *thd, List<Item> &update_fields, List<Item> &update_values, Item *conds);
-    bool check_fast_update(THD *thd, List<Item> &update_fields, List<Item> &update_values, Item *conds);
-    int send_update_message(List<Item> &update_fields, List<Item> &update_values, Item *conds, DB_TXN *txn);
-    int upsert(THD *thd, List<Item> &update_fields, List<Item> &update_values);
-    bool check_upsert(THD *thd, List<Item> &update_fields, List<Item> &update_values);
-    int send_upsert_message(THD *thd, List<Item> &update_fields, List<Item> &update_values, DB_TXN *txn);
-#endif
+    int fast_update(THD *thd,
```

Comment:
Every single one of these should probably be MY_ATTRIBUTE((warn_unused_result))

Reply:
I think, this is not a good idea to use compiler-specific keywords without strict necessity.

Reply:
It will give better diagnostics and will be defined to nothing on unsupporting compilers. No functionality or portability is lost, and the code base already contains many of them.

Reply:
Basically every function whose return value should be checked should carry it, so I agree re. including in the codestyle guide. So I agree that it should be added to our codestyle guide. Those sample cases should have added it too, but missed it. I plan to do a batch add on e.g. 8.0 port.

Reply:
This can work, but I am not sure whether existing uses in our patch should be converted, which would seem to be a refactoring that is slightly hard to justify.

Reply:
Following this logic, if we are going to include using of MY_ATTRIBUTE((warn_unused_result)) in our code style sheet, then refactoring of our existing code is required too.

I think we could use the same rule as we use for indentation. Currently we reindent only the code which is fixed for our purposes.

Reply:
I agree